packages/neuron-ui/src/components/PageContainer/hooks.ts

```diff
@@ -0,0 +1,109 @@
+import { useCallback, useEffect, useState } from 'react'
+import { importedWalletDialogShown } from 'services/localCache'
+import { isDark, openExternal, updateWallet, setTheme as setThemeAPI } from 'services/remote'
+import { Migrate } from 'services/subjects'
+import { getExplorerUrl, isSuccessResponse } from 'utils'
+
+export const useSetBlockNumber = ({
+  firstAddress,
+  walletID,
+  isMainnet,
+  isLightClient,
+  isHomePage,
+}: {
+  firstAddress: string
+  walletID: string
+  isMainnet: boolean
+  isLightClient: boolean
+  isHomePage?: boolean
+}) => {
+  const [showSetStartBlock, setShowSetStartBlock] = useState(false)
+  const [startBlockNumber, setStartBlockNumber] = useState('')
+  const [blockNumberErr, setBlockNumberErr] = useState('')
+  const onChangeStartBlockNumber = useCallback((e: React.SyntheticEvent<HTMLInputElement>) => {
+    const { value } = e.currentTarget
+    const blockNumber = value.replace(/,/g, '')
+    if (Number.isNaN(+blockNumber) || /[^\d]/.test(blockNumber)) {
+      return
+    }
+    setStartBlockNumber(blockNumber)
+    setBlockNumberErr('')
+  }, [])
+  const onOpenAddressInExplorer = useCallback(() => {
+    const explorerUrl = getExplorerUrl(isMainnet)
+    openExternal(`${explorerUrl}/address/${firstAddress}`)
+  }, [firstAddress])
+  const onViewBlock = useCallback(() => {
+    const explorerUrl = getExplorerUrl(isMainnet)
+    openExternal(`${explorerUrl}/block/${startBlockNumber}`)
+  }, [startBlockNumber, isMainnet])
+  const onConfirm = useCallback(() => {
+    updateWallet({ id: walletID, startBlockNumberInLight: `0x${BigInt(startBlockNumber).toString(16)}` }).then(res => {
```

Comment:
`startBlockNumberInLight` can be simplified to `startBlockNumber` because the on-chain data are the same no matter full node/client is connected to.

And naming it `startBlockNumber` avoids further change when data of full node can be compressed